packages/api/tests/endpoints/test-granules-updateGranulesAndFilesCollection.js

```diff
@@ -0,0 +1,339 @@
+/* eslint-disable no-await-in-loop */
+const test = require('ava');
+const range = require('lodash/range');
+
+const cryptoRandomString = require('crypto-random-string');
+const {
+  CollectionPgModel,
+  destroyLocalTestDb,
+  fakeCollectionRecordFactory,
+  fakeFileRecordFactory,
+  fakeGranuleRecordFactory,
+  FilePgModel,
+  generateLocalTestDb,
+  getUniqueGranuleByGranuleId,
+  GranulePgModel,
+  localStackConnectionEnv,
+  migrationDir,
+  translatePostgresCollectionToApiCollection,
+  translatePostgresGranuleResultToApiGranule,
+} = require('@cumulus/db');
+const {
+  createBucket,
+  recursivelyDeleteS3Bucket,
+  s3PutObject,
+} = require('@cumulus/aws-client/S3');
+const { createSnsTopic } = require('@cumulus/aws-client/SNS');
+const indexer = require('@cumulus/es-client/indexer');
+const {
+  createTestIndex,
+  cleanupTestIndex,
+} = require('@cumulus/es-client/testUtils');
+const { randomString, randomId } = require('@cumulus/common/test-utils');
+const { constructCollectionId } = require('@cumulus/message/Collections');
+const models = require('../../models');
+
+const { request } = require('../helpers/request');
+
+// Dynamo mock data factories
+const {
+  createFakeJwtAuthToken,
+  setAuthorizedOAuthUsers,
+} = require('../../lib/testUtils');
+
+const testDbName = `granules_${cryptoRandomString({ length: 10 })}`;
+
+let accessTokenModel;
+let jwtAuthToken;
+
+process.env.AccessTokensTable = randomId('token');
+process.env.stackName = randomId('stackname');
+process.env.system_bucket = randomId('system-bucket');
+process.env.TOKEN_SECRET = randomId('secret');
+process.env.backgroundQueueUrl = randomId('backgroundQueueUrl');
+
+// import the express app after setting the env variables
+const { app } = require('../../app');
+
+/**
+ * Simulate granule records post-collection-move for database updates test
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction - DB client or transaction
```

Comment:
nit: this is missing the knex param....but.... *probably* in ava tests we can skip the type annotations for the params.  We've not adopted attempting typing in these for a variety of reasons both good and bad.